src/Snapshot.js

```diff
@@ -84,41 +68,47 @@ const getArrayOfParents = path => {
   return result;
 };
 
-const buildName: (
-  snapshotNode: Node,
-  parents: Array<Node>,
-  position: number,
-) => string = (snapshotNode, parents, position) => {
+const buildName: (snapshotNode: Node, parents: Array<Node>, position: number) => string = (
+  snapshotNode,
+  parents,
+  position
+) => {
   const fullName = parents.map(parent => parent.arguments[0].value).join(' ');
 
   return utils.testNameToKey(fullName, position);
 };
 
 export default class Snapshot {
   _parser: Function;
+
   _matchers: Array<string>;
+
   _projectConfig: ?ProjectConfig;
-  constructor(
-    parser: any,
-    customMatchers?: Array<string>,
-    projectConfig?: ProjectConfig,
-  ) {
+
+  constructor(parser: any, customMatchers?: Array<string>, projectConfig?: ProjectConfig) {
     this._parser = parser || getASTfor;
-    this._matchers = ['toMatchSnapshot', 'toThrowErrorMatchingSnapshot'].concat(
-      customMatchers || [],
-    );
+    this._matchers = ['toMatchSnapshot', 'toThrowErrorMatchingSnapshot'].concat(customMatchers || []);
     this._projectConfig = projectConfig;
   }
 
-  getMetadata(filePath: string): Array<SnapshotMetadata> {
-    const fileNode = this._parser(filePath);
+  getMetadata(filePath: string, verbose: boolean = false): Array<SnapshotMetadata> {
+    let fileNode;
+    try {
+      fileNode = this._parser(filePath);
+    } catch (error) {
+      if (verbose) {
+        // eslint-disable-next-line no-console
+        console.warn(error);
+      }
+      return [];
+    }
     const state = {
       found: [],
     };
     const Visitors = {
-      Identifier(path, state, matchers) {
+      Identifier(path, _state, matchers) {
         if (matchers.indexOf(path.node.name) >= 0) {
-          state.found.push({
+          _state.found.push({
```

Comment:
Is it just me, or does this object assignment with the `Identifier` function look a bit off? It took a little bit to realize it's a function declaration, and sort out the `_state` vs `state`. With that in mind, how about we clean up this painful block a bit:

Here's what we've got:
```js
    const Visitors = {
      Identifier(path, _state, matchers) {
        if (matchers.indexOf(path.node.name) >= 0) {
          _state.found.push({
            node: path.node,
            parents: getArrayOfParents(path),
          });
        }
      },
    };

    traverse(fileNode, {
      enter: path => {
        const visitor = Visitors[path.node.type];
        if (visitor != null) {
          visitor(path, state, this._matchers);
        }
      },
    });
```

It looks like we're trying to add a node and parents to the `state.found` array if:
* The node is an identifier (`node.path.type === 'Identifier'`)
* It is one of the built-in or custom matchers in the `this._matchers` list

If we were to take these two criteria into a private class method:
```js
  _isSnapshotMatcher(path) {
    return path && path.node && path.node.type === 'Identifier'
      && this._matchers.indexOf(path.node.name) >= 0;
  }
```
then we can reduce that whole block to:
```js
    traverse(fileNode, {
      enter: path => {
        if (this._isSnapshotMatcher(path)) {
          state.found.push({
            node: path.node,
            parents: getArrayOfParents(path),
          });
        }
      },
    });
```

<br/>

If you really wanted to go overboard, why do we need the `.` in `state.found` if if doesn't look like we're using any other members on the `state` object?

Reply:
I definitely agree with you this code can be made more readable, but it has been like this for a while, all I did is to rename `state` to `_state` to conform to the lint rule "no-shadow"...  are you sure we want to do the refactoring here, given the PR is already large and this is not related to what we try to fix?

Reply:
Hah! Thanks for calling me on the scope creep. Your change is AOK. 👌
I'll open a new PR .. someday. 😜